Packs/NozomiNetworks/Integrations/NozomiNetworks/NozomiNetworks.py

```diff
@@ -0,0 +1,447 @@
+        lft, \
+        lfid
+
+
+def last_fetched_time(inc, last_run):
+    return inc[-1]['time'] if len(inc) > 0 else last_run.get("last_fetch", 0)
+
+
+def last_fetched_id(inc, last_run):
+    return inc[-1]['id'] if len(inc) > 0 else last_run.get("last_id", None)
+
+
+def last_asset_id(response):
+    return response[-1]['id'] if len(response) > 0 else ''
+
+
+def ack_unack_alerts(ids, status, client):
+    data = []
+    for id in ids:
+        data.append({'id': id, 'ack': status})
+    response = client.http_post_request('/api/open/alerts/ack', {'data': data})
+    return response["result"]["id"]
+
+
+def ack_alerts(ids, client):
+    return ack_unack_alerts(ids, True, client)
+
+
+def nozomi_alerts_ids_from_demisto_incidents(demisto_incidents):
+    return ids_from_incidents([json.loads(incident['rawJSON']) for incident in demisto_incidents])
+
+
+def close_alerts(args, close_action, client):
+    readable_close_action = "closed_as_security" if close_action == "delete_rules" else "closed_as_change"
+    extracted_ids = argToList(args.get('ids'))
+    human_readable = f'Command changes the status of the following alerts: {extracted_ids} ' \
+        f'passed as "{readable_close_action}" in Nozomi Networks platform.'
+
+    client.http_post_request(
+        '/api/open/alerts/close',
+        {"ids": extracted_ids, "close_action": close_action})
+
+    return {
+        'readable_output': human_readable,
+        'outputs_prefix': None,
+        'outputs_key_field': None,
+        'outputs': None
+    }
+
+
+def has_last_id(lr):
+    return lr is not None and 'last_id' in lr
+
+
+def get_last_id(last_run):
+    result = None
+    if has_last_id(last_run):
+        result = f'{last_run.get("last_id", 0)}'
+    return result
+
+
+def filter_from_args(args):
+    if args and args.get('filter', '') != '':
+        filter = args.get('filter', '')
+        if '| where' in filter:
+            return filter
+        else:
+            return f" | where {filter}"
+    else:
+        return ''
+
+
+def assets_limit_from_args(args):
+    if args:
+        limit = int(args.get('limit', DEFAULT_ASSETS_FINDABLE_BY_A_COMMAND))
+        if limit > MAX_ASSETS_FINDABLE_BY_A_COMMAND:
+            return MAX_ASSETS_FINDABLE_BY_A_COMMAND
+        else:
+            return limit
+    else:
+        return DEFAULT_ASSETS_FINDABLE_BY_A_COMMAND
+
+
+def nodes_confirmed_filter(only_nodes_confirmed):
+    if only_nodes_confirmed and only_nodes_confirmed == 'True':
+        return ' | where mac_address:info.likelihood_level == confirmed'
+    else:
+        return ''
+
+
+def humanize_api_error(error):
+    if '401' in error:
+        return 'Authentication error, check your username and password'
+    else:
+        return error
+
+
+''' MAIN_FUNCTION '''
+
+
+def fetch_incidents(
+        client,
+        st=None,
+        last_run=None,
+        last_id=None,
+        risk=None,
+        fetch_also_n2os_incidents=None,
+        test_mode=False
```

Comment:
Why is this needed?

Reply:
To test the function with different values

Reply:
But I don't see we use this anywhere. Plus this values will always come from the user no? Which is exactly what is happening in the following lines.

Reply:
I have integration test, not sent to the PR cause they need a running instance of nozomi to run.
Something like that:

`def test_fetch_incidents_without_n2os_incidents():
    i, last_fetch = fetch_incidents(
        __real_client(),
        st='1392040975219',
        risk='1',
        fetch_also_n2os_incidents=False,
    )
    assert last_fetch is not None
    assert len(i) == 16`